monitoring/bin/create_elasticsearch_datasource.sh

```diff
@@ -0,0 +1,203 @@
+   grfds_user="${nst}_grafana_ds"
+fi
+
+if user_exists "$grfds_user"; then
+   log_verbose "Removing the existing [$grfds_user] utility account."
+   delete_user $grfds_user
+fi
+
+grfds_passwd="$(randomPassword)"
+
+if [ "$cluster" == "true" ]; then
+   ./logging/bin/user.sh CREATE -ns _all_ -t _all_ -u $grfds_user -p "$grfds_passwd" -g
+elif [ -z "$tenant" ]; then
+   ./logging/bin/user.sh CREATE -ns $tenantNS -u $grfds_user -p "$grfds_passwd" -g
+else
+   ./logging/bin/user.sh CREATE -ns $tenantNS -t $tenant -u $grfds_user -p "$grfds_passwd" -g
+fi
+
+# Create temporary directory for string replacement in the grafana-datasource-es.yaml file
+monDir=$TMP_DIR/$MON_NS
+mkdir -p $monDir
+cp monitoring/grafana-datasource-es.yaml $monDir/grafana-datasource-es.yaml
+
+# Replace placeholders
+log_debug "Replacing variables in $monDir/grafana-datasource-es.yaml file"
+if echo "$OSTYPE" | grep 'darwin' > /dev/null 2>&1; then
+    sed -i '' "s/__namespace__/$LOG_NS/g" $monDir/grafana-datasource-es.yaml
+    sed -i '' "s/__userID__/$grfds_user/g" $monDir/grafana-datasource-es.yaml
+    sed -i '' "s/__passwd__/$grfds_passwd/g" $monDir/grafana-datasource-es.yaml
+else
+    sed -i "s/__namespace__/$LOG_NS/g" $monDir/grafana-datasource-es.yaml
+    sed -i "s/__userID__/$grfds_user/g" $monDir/grafana-datasource-es.yaml
+    sed -i "s/__passwd__/$grfds_passwd/g" $monDir/grafana-datasource-es.yaml
+fi
+
+# Removes old Elasticsearch data source if one exists
+if [ "$cluster" == "true" ]; then
+    if [[ -n "$(kubectl get secret -n $MON_NS grafana-datasource-es -o custom-columns=:metadata.name --no-headers --ignore-not-found)" ]]; then
+        log_info "Removing existing Elasticsearch data source ..."
+        kubectl delete secret -n $MON_NS --ignore-not-found grafana-datasource-es
+    fi
+else
+    if [ -n "$(kubectl get secret -n $tenantNS v4m-grafana-datasource-es-$tenant -o custom-columns=:metadata.name --no-headers --ignore-not-found)" ]; then
+        log_info "Removing existing Elasticsearch data source for [$tenantNS/$tenant] ..."
+        kubectl delete secret -n $tenantNS --ignore-not-found v4m-grafana-datasource-es-$tenant
+    fi
+fi
+
+# Adds the Elasticsearch data source to Grafana
+log_info "Provisioning Elasticsearch datasource for Grafana"
+if [ "$cluster" == "true" ]; then
+    kubectl create secret generic -n $MON_NS grafana-datasource-es --from-file $monDir/grafana-datasource-es.yaml
+    kubectl label secret -n $MON_NS grafana-datasource-es grafana_datasource=1 sas.com/monitoring-base=kube-viya-monitoring
+else
+    kubectl create secret generic -n $tenantNS v4m-grafana-datasource-es-$tenant --from-file $monDir/grafana-datasource-es.yaml
+    kubectl label secret -n $tenantNS v4m-grafana-datasource-es-$tenant grafana_datasource-$tenant=true sas.com/monitoring-base=kube-viya-monitoring
+fi
+
+# Delete pods so that they can be restarted with the change.
+log_info "Elasticsearch data source provisioned.  Restarting pods to apply the change"
```

Comment:
Do we really need to kill the pods to have the newly provisioned datasource available?  Seems like the datasource provisioning sidecar should be able to do what is needed without having us manually deleting the pod.

Reply:
Unfortunately, we did need to kill the pods in order for the change to take effect.  When I was making the scripts initially, the Grafana UI would not pick up the change unless the pods were restarted.